### PR TITLE
SONARJAVA-2360 Fix CFG handling of continue/break nested in try/finally

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/cfg/CFG.java [hunk 1]
```diff
@@ -700,6 +700,15 @@ private void buildBreakStatement(BreakStatementTree tree) {
       targetBlock = labelsBreakTarget.get(label.name());
     }
     currentBlock = createUnconditionalJump(tree, targetBlock);
+    if(currentBlock.exitBlock != null) {
+      currentBlock.exitBlock = null;
+      Block last = breakTargets.removeLast();
+      targetBlock.successors().clear();
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/cfg/CFG.java]: MAJOR A "NullPointerException" could be thrown; "targetBlock" is nullable here. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java [hunk 2]
```diff
@@ -937,6 +950,65 @@ public void while_loop_with_continue() {
...
+        element(IDENTIFIER, "foo"),
+        element(METHOD_INVOCATION)
+      ).terminator(WHILE_STATEMENT).successors(0, 4),
+      block(element(TRY_STATEMENT)).successors(3),
+      block(
+        element(Tree.Kind.IDENTIFIER, "bar"),
+        element(STRING_LITERAL, "try"),
+        element(METHOD_INVOCATION)
+      ).successors(2).exceptions(1),
+      new BlockChecker(BREAK_STATEMENT, 1).hasNoExitBlock(),
```
Comment (m-g-sonar) [inline java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java]: I would prefer having `terminator(BREAK_STATEMENT, 1).hasNoExitBlock()` here instead, in order to stay closer from the syntax used in other tests.

File: java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java [hunk 3]
```diff
@@ -937,6 +950,65 @@ public void while_loop_with_continue() {
...
+        element(IDENTIFIER, "foo"),
+        element(METHOD_INVOCATION)
+      ).terminator(WHILE_STATEMENT).successors(0, 4),
+      block(element(TRY_STATEMENT)).successors(3),
+      block(
+        element(Tree.Kind.IDENTIFIER, "bar"),
+        element(STRING_LITERAL, "try"),
+        element(METHOD_INVOCATION)
+      ).successors(2).exceptions(1),
+      new BlockChecker(CONTINUE_STATEMENT, 1).hasNoExitBlock(),
```
Comment (m-g-sonar) [inline java-frontend/src/test/java/org/sonar/java/cfg/CFGTest.java]: I would prefer having `terminator(CONTINUE_STATEMENT, 1).hasNoExitBlock()` here instead, in order to stay closer from the syntax used in other tests.

File: java-frontend/src/main/java/org/sonar/java/cfg/CFG.java (hunk 1 above)
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/cfg/CFG.java]: The NPE reported here is a FP caused by learning the NULL constraint from method invocation `createUnconditionalJump(Tree, @Nullable Block)` [line 702] which makes the engine learn that target block may be nullable.

Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/cfg/CFG.java]: As dicussed, this does not require a fix in this PR. However shall this lead to open a ticket on the subject ?

**Pull request merged**